### PR TITLE
Introducing estimated feature(s) transformer on paired data

### PR DESCRIPTION
**Pull request by natsheh**

### CODE REVIEW

File: beard/similarity/pairs.py [hunk 1]
```diff
@@ -229,6 +231,118 @@ def transform(self, X):
...
+        -------
+        :returns Xt: array-like, shape (n_samples, 2 * n_features_prime)
+            The transformed data.
+        """
+        Xt = self.classifier.decision_function(X)
+        return Xt
+
+
+def sigmoid(x):
+    """Sigmoid function."""
```
Comment (MSusik) [inline beard/similarity/pairs.py]: http://docs.scipy.org/doc/scipy/reference/generated/scipy.special.expit.html

File: beard/similarity/pairs.py [hunk 2]
```diff
@@ -229,6 +231,118 @@ def transform(self, X):
...
+            Input data.
+
+        Returns
+        -------
+        :returns: self
+        """
+        return self
+
+    def transform(self, X, y=None):
+        """Compute the cosine similarity for all pairs of elements in ``X``.
```
Comment (MSusik) [inline beard/similarity/pairs.py]: Is the docstring OK?

File: beard/similarity/pairs.py [hunk 3]
```diff
@@ -229,6 +231,114 @@ def transform(self, X):
         return Xt.reshape((n_samples, 1))
 
 
+class EthnicityTransformer(TransformerMixin):
```
Comment (MSusik) [inline beard/similarity/pairs.py]: It looks much more generic than just an ethnicity transformer.

File: beard/similarity/pairs.py [hunk 4]
```diff
@@ -229,6 +231,113 @@ def transform(self, X):
         return Xt.reshape((n_samples, 1))
 
 
+class EstimatedFeature(TransformerMixin):
```
Comment (glouppe) [inline beard/similarity/pairs.py]: Can you rebrand this as `EstimatorTransformer` and rename `classifier` to `estimator` everywhere in the class?

File: beard/similarity/pairs.py [hunk 5]
```diff
@@ -229,6 +231,113 @@ def transform(self, X):
...
+
+    """Estimated feature transformer on paired data."""
```
Comment (glouppe) [inline beard/similarity/pairs.py]: Rewrite as: "Wrap an estimator `decision_function` as a `transform` method."

File: beard/similarity/pairs.py [hunk 6]
```diff
@@ -229,6 +231,113 @@ def transform(self, X):
...
+
+    def __init__(self, classifier):
+        """Initialize with the feature estimator.
+
+        Parameters
+        ----------
+        :param classifier: Feature estimator that takes paired input data and
```
Comment (glouppe) [inline beard/similarity/pairs.py]: Rewrite simply as "estimator trained on paired data."

**Comment by glouppe** [comment]
Please add tests for this.

File: beard/similarity/pairs.py [hunk 7]
```diff
@@ -229,6 +231,113 @@ def transform(self, X):
...
+        :returns Xt: array-like, shape (n_samples, 2 * n_features_prime)
+            The transformed data.
+        """
+        Xt = self.classifier.decision_function(X)
+        return Xt
+
+
+class ElementMultiplication(TransformerMixin):
+
+    """Element multiplication on paired data."""
```
Comment (glouppe) [inline beard/similarity/pairs.py]: Should be "Element-wise" multiplication

File: beard/similarity/pairs.py [hunk 8]
```diff
@@ -229,6 +231,113 @@ def transform(self, X):
...
+            Input data.
+
+        Returns
+        -------
+        :returns: self
+        """
+        return self
+
+    def transform(self, X):
+        """Compute the element multiplication for all pairs of elements in ``X``.
```
Comment (glouppe) [inline beard/similarity/pairs.py]: element-wise

File: beard/similarity/pairs.py [hunk 9]
```diff
@@ -229,6 +230,110 @@ def transform(self, X):
...
+        :returns Xt: array-like, shape (n_samples, 2 * n_features_prime)
+            The transformed data.
+        """
+        Xt = self.estimator.decision_function(X)
+        return Xt
+
+
+class ElementMultiplication(TransformerMixin):
+
+    """Element-wise multiplication on the sigmoid of paired data."""
```
Comment (glouppe) [inline beard/similarity/pairs.py]: you forgot to update this docstring

**Comment by natsheh** [comment]
ready for reviewing

**Comment by glouppe** [comment]
Good to go. Merging.

**Pull request merged**